### PR TITLE
Remove packages that are not whitelisted

### PR DESCRIPTION
**Pull request by zfy0701**

### CODE REVIEW

File: src/dependency-manager.ts [hunk 1]
```diff
@@ -28,37 +30,109 @@ export class DependencyManager {
...
+                    await this.rimrafAysnc(childPath, {});
+                  } catch (e) {
+                    this.logger.error('Deleting repo error: ' +  e.toString());
+                  }
+                }
+              }
+            } catch (e) {
+              this.logger.log(e);
+            }
+            const nodeModulesDir = resolve(childPath, 'node_modules');
```
Comment (Poytr1) [inline src/dependency-manager.ts]: If the childPath is not in our whitelist, the childPath has been deleted before, and we needn't try to look into node_modules inside it(all of them should be deleted too)

File: src/dependency-manager.ts [hunk 2]
```diff
@@ -28,37 +30,109 @@ export class DependencyManager {
...
+      promise.childProcess.stderr.on('data', (data) => {
+        if (totalLength < MAX_YARN_OUTPUT) {
+          const msg = data.toString();
+          this.logger.info('[yarn]', msg);
+          totalLength += msg.length;
+        }
+      });
+      await promise;
+    } catch (e) {
+      this.logger.error('Can\'t launch yarn to download dependencies');
```
[the remark this answers is not in the document]
Comment (zfy0701) [reply src/dependency-manager.ts]: done

File: src/dependency-manager.ts (hunk 1 above)
Comment (zfy0701) [reply src/dependency-manager.ts]: done

File: src/dependency-manager.ts [hunk 3]
```diff
@@ -28,37 +30,109 @@ export class DependencyManager {
...
-    //     console.debug('stderr:' + data)
-    // })
-    //
-    // this.npmProcess.on('error', err => {
-    //     console.debug('error:' + err)
-    // })
+      promise.childProcess.stderr.on('data', (data) => {
+        if (totalLength < MAX_YARN_OUTPUT) {
+          const msg = data.toString();
+          this.logger.info('[yarn]', msg);
```
[the remark this answers is not in the document]
Comment (zfy0701) [reply src/dependency-manager.ts]: intended to just ignore warning from yarn process, doesn't really matter to the rest

**Pull request merged**